int32 msg_to_nodes
int32 sub_index
int32 node_num
